src/App.tsx

```diff
@@ -15,36 +23,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortBy {
+  'default',
+  'alpabet',
+  'length',
+}
+
+function getGoods(goodsFromServList: string[]): GoodsList {
+  return goodsFromServList.map((str, index) => ({ id: index, val: str }));
+}
+
+function prepGoods(goods: GoodsList, oper: SortBy, revOp: boolean): GoodsList {
+  let tempArr = getGoods(goodsFromServer);
+
+  if (oper === 0) {
+    return revOp ? tempArr.reverse() : tempArr;
+  }
+
+  tempArr = goods.sort((g1: GoodsFromServer, g2: GoodsFromServer) => {
+    switch (oper) {
+      case 1:
+        return g1.val.localeCompare(g2.val);
+      case 2:
+        return g1.val.length - g2.val.length;
+      default:
+        return 0;
+    }
+  });
+
+  return revOp ? tempArr.reverse() : tempArr;
+}
+
+export const App = () => {
+  const [sortBy, setSortBy] = useState<SortBy>(0);
+  const [rev, setRev] = useState<boolean>(false);
+
+  const visGoods = prepGoods(getGoods(goodsFromServer), sortBy, rev);
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortBy !== 1,
+          })}
+          onClick={() => setSortBy(1)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortBy !== 2,
+          })}
+          onClick={() => setSortBy(2)}
```

Comment:
[UX] - Similar to the previous comment, the active button for sorting by length is not marked. Ensure that the user can easily tell which sorting method is active by adding appropriate styling.